cpp/src/parquet/geospatial_statistics_test.cc

```diff
@@ -0,0 +1,317 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "arrow/array/builder_binary.h"
+#include "arrow/compute/api.h"
+#include "arrow/testing/gtest_util.h"
+
+#include "parquet/geospatial_statistics.h"
+#include "parquet/test_util.h"
+
+namespace parquet::geometry {
+
+TEST(TestGeoStatistics, TestDefaults) {
+  GeoStatistics stats;
+  EXPECT_EQ(stats.geometry_types().size(), 0);
+  EXPECT_TRUE(stats.is_valid());
+  EXPECT_TRUE(stats.is_empty());
+  EXPECT_FALSE(stats.has_x());
+  EXPECT_FALSE(stats.has_y());
+  EXPECT_FALSE(stats.has_z());
+  EXPECT_FALSE(stats.has_m());
+  EXPECT_EQ(stats.xmax() - stats.xmin(), -kInf);
+  EXPECT_EQ(stats.ymax() - stats.ymin(), -kInf);
+  EXPECT_EQ(stats.zmax() - stats.zmin(), -kInf);
+  EXPECT_EQ(stats.mmax() - stats.mmin(), -kInf);
+  EXPECT_TRUE(stats.Equals(GeoStatistics()));
+
+  EXPECT_EQ(stats.ToString(),
+            "GeoStatistics \n  x: [inf, -inf]\n  y: [inf, -inf]\n  geometry_types:\n");
+
+  auto encoded = stats.Encode();
+  EXPECT_FALSE(encoded.is_set());
+  EXPECT_FALSE(encoded.has_x());
+  EXPECT_FALSE(encoded.has_y());
+  EXPECT_FALSE(encoded.has_z());
+  EXPECT_FALSE(encoded.has_m());
+  EXPECT_TRUE(GeoStatistics(encoded).Equals(stats));
```

Comment:
Can we test `Equals` being false sometimes? Otherwise an incomplete implementation of `Equals` would pass the tests.